backend/api/Controllers/AreaController.cs

```diff
@@ -271,6 +271,35 @@ public async Task<ActionResult<AreaResponse>> GetAreaById([FromRoute] string id)
             }
         }
 
+        /// <summary>
+        /// Lookup area by specified id.
+        /// </summary>
+        [HttpGet]
+        [Authorize(Roles = Role.Any)]
+        [Route("deck/{deckId}")]
+        [ProducesResponseType(typeof(AreaResponse), StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        public async Task<ActionResult<IList<AreaResponse>>> GetAreaByDeckId([FromRoute] string deckId)
+        {
+            try
+            {
+                var areas = await _areaService.ReadByDeckId(deckId);
+                if (!areas.Any())
+                    return NotFound($"Could not find area for deck with id {deckId}");
```

Comment:
This causes quite a few 404 errors as not all decks have an area. Should we change this or do we assume that all decks should have at least one area?

Reply:
I think at the moment deck doesn't even have an area so to avoid this we would have to database changes which I don't think should be for this PR